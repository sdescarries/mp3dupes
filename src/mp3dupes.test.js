const mp3dupes = require('./mp3dupes');
const fs = require('fs-extra');

console.log = jest.fn();
console.warn = jest.fn();
console.error = jest.fn();
process.exit = jest.fn();

describe('mp3dupes', () => {

  beforeAll(() => {

    console.log({
      __dirname,
      __filename,
      argv: process.argv,
    });

    // copy test data in volatile directory for test input
    fs.ensureDirSync('./results/');
    fs.removeSync('./results/mp3dupes');
    fs.copySync('./data', './results/mp3dupes');

  });
  /*
  it('resolves --help', async (done) => {
    process.argv = ['node', 'mp3dupes', '--help'];
    await expect(mp3dupes()).resolves.toBeUndefined();
    done();
  });
*/
  it('resolves test data', async (done) => {
    process.argv = ['node', 'mp3dupes', '--silent', '--rename', './results/mp3dupes'];
    await expect(mp3dupes()).resolves.toBeUndefined();
    done();
  });

});
