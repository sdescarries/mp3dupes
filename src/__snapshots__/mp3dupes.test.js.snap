// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mp3dupes renames and removes duplicates 1`] = `
Array [
  "001 Wait It Out Artec.mp3",
  "002 Winter ENiGMA Dubz.mp3",
  "003 Im Lost A lEte Passe.mp3",
  "004 Foo Bar.mp3",
  "005 I Dont Feel It AsscDhE nouya.mp3",
  "plain text.txt",
]
`;

exports[`mp3dupes resolves help with no change 1`] = `
Array [
  "123 Foo Bar.mp3",
  "123 duplicate B.mp3",
  "234 duplicate C.mp3",
  "32.mp3",
  "412 duplicate A.mp3",
  "53 unknown.mp3",
  "no sequence.mp3",
  "no tags.mp3",
  "plain text.txt",
]
`;
